extensions/ipynb/src/ipynbMain.ts

```diff
@@ -21,7 +21,20 @@ export function activate(context: vscode.ExtensionContext) {
 	return {
 		exportNotebook: (notebook: vscode.NotebookData): string => {
 			return exportNotebook(notebook, serializer);
-		}
+		},
+		setKernelSpec: async (resource: vscode.Uri, kernelspec: unknown): Promise<boolean> => {
+			const document = vscode.workspace.notebookDocuments.find(doc => doc.uri.toString() === resource.toString());
+			if (!document) {
+				return false;
+			}
+
+			const edit = new vscode.WorkspaceEdit();
+			edit.replaceNotebookMetadata(resource, {
+				...document.metadata,
+				kernelspec: kernelspec
+			});
```

Comment:
The metadata isn't stored at the root level, to my knowledge it's stirred in `metadata.custom`.
Also I don't think this will get saved into the ipynb file.
We use custom, as that's where custom metada had to be stored  (later changed in vscode api).